workflows/prognostic_c48_run/runtime/steppers/machine_learning.py

```diff
@@ -200,9 +227,17 @@ def __call__(self, time, state):
         dQ1_initial = tendency.get("dQ1", xr.zeros_like(state[SPHUM]))
         dQ2_initial = tendency.get("dQ2", xr.zeros_like(state[SPHUM]))
 
-        dQ1_updated, dQ2_updated = non_negative_sphum(
-            state[SPHUM], dQ1_initial, dQ2_initial, dt=self.timestep,
-        )
+        if self.mse_conserving_limiter:
```

Comment:
Could this routine be wrapped into a single `moisture_conserving_limiter` function that takes in `sphum`, `dQ2`, `dt`, `Optional[dQ1]` and returns a dataset of updated `dQ2` and optionally `dQ1`? The point being that the same function could be called from both the machine learning stepper and the fv3fit step transformer, which seems easier to track and test.

Reply:
Done.